src/app/services/condition/condition.service.ts

```diff
@@ -0,0 +1,20 @@
+import {Injectable} from '@angular/core';
+import {Subject, Subscription} from "rxjs";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ConditionService {
+  private subject$ = new Subject();
+
+  constructor() {
+  }
+
+  emit(condition: boolean) {
```

Comment:
Emit and subscribe are useless wrappers over the subject. You can give a more meaningful name to the subject and expose it as a public method, that'll accomplish the same goal. To MovieDetailsService and FindMovieService it applies too.